src/main/java/org/sonarlint/intellij/config/global/SonarLintGlobalSettings.java

```diff
@@ -47,13 +48,52 @@
   private boolean autoTrigger = true;
   private List<SonarQubeServer> servers = new LinkedList<>();
   private List<String> fileExclusions = new LinkedList<>();
-  private Set<String> includedRules = new HashSet<>();
-  private Set<String> excludedRules = new HashSet<>();
+  @Deprecated
+  private final Set<String> includedRules = new HashSet<>();
+  @Deprecated
+  private final Set<String> excludedRules = new HashSet<>();
+  private final Map<String, Rule> rules = new HashMap<>();
 
   public static SonarLintGlobalSettings getInstance() {
     return ApplicationManager.getApplication().getComponent(SonarLintGlobalSettings.class);
   }
 
+  public void setRuleParam(String ruleKey, String paramName, String paramValue) {
+    if (!rules.containsKey(ruleKey)) {
```

Comment:
you can use computeIfAbsent here

Reply:
I believe we need a `resetRuleParam(String ruleKey, String paramName)` method, to remove the explicitly set value in case a rule is "reset" to its default behavior.